ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/reader/implementation/file/FileReaderByPath.java

```diff
@@ -0,0 +1,57 @@
+package edu.usb.argos.ASTProcessor.reader.implementation.file;
+
+import edu.usb.argos.ASTProcessor.antlr.JavaLexer;
+import edu.usb.argos.ASTProcessor.antlr.JavaParser;
+import edu.usb.argos.ASTProcessor.reader.exceptions.FileAnalyzerException;
+import edu.usb.argos.ASTProcessor.reader.exceptions.ParserException;
+import edu.usb.argos.ASTProcessor.reader.interfaces.IFileAnalyzer;
+import edu.usb.argos.ASTProcessor.reader.interfaces.IFileValidationStrategy;
+import edu.usb.argos.ASTProcessor.reader.validations.PathValidationStrategy;
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.CharStreams;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.antlr.v4.runtime.CommonTokenStream;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class FileReaderByPath implements IFileAnalyzer<Path, ParseTree> {
+    private final IFileValidationStrategy<Path> validationStrategy;
+
+    public FileReaderByPath() {
+        this.validationStrategy = new PathValidationStrategy();
+    }
+
+    public FileReaderByPath(IFileValidationStrategy<Path> validationStrategy) {
+        this.validationStrategy = validationStrategy;
+    }
+
+    @Override
+    public ParseTree readFile(Path codePath) throws FileAnalyzerException {
```

Comment:
Instead of throwing exceptions, can you handle them and show logs?

In my [Source Tree Analyzer Implementation Branch](https://github.com/SoftDevSix/Argos/pull/7) I already did it using Slf4 as Lombok annotation, you can bring my changes to your branch to work with that base.

[click here for more information about the logger](https://medium.com/@AlexanderObregon/enhancing-logging-with-log-and-slf4j-in-spring-boot-applications-f7e70c6e4cc7)

Reply:
we should log exceptions and throw them

Reply:
Done